update: using IEquatable with serialization (#1475)
Unity-Technologies/com.unity.multiplayer.docs: docs/advanced-topics/serialization/inetworkserializable.md

    {
        OnNetworkSerialize(serializer);
    }
}

public class MyExtendedData : MyBaseClass, IEquatable<MyExtendedData>
{
    public long Data;
    public bool Equals(MyExtendedData other)
    {
        // Compare at the child's level and the base class level
        return Data.Equals(other.Data) && base.Equals(other);
    }

    protected override void OnNetworkSerialize<T>(BufferSerializer<T> serializer)
    {
        // This serializes in an asscending order (child to root parent)
        serializer.SerializeValue(ref Data);
        base.OnNetworkSerialize(serializer);

        // This serializes in a descending order (root parent to last generation child)
        //base.OnNetworkSerialize(serializer);
        //serializer.SerializeValue(ref Data);
    }
}

public class MyBehaviour : NetworkBehaviour
{
    public NetworkVariable<MyExtendedData> SomeNetVar = new NetworkVariable<MyExtendedData>();
}
```

Where `MyExtendedData` implements `IEquatable<MyExtendedData>` and overriding the `OnNetworkSerialize` allows one to stack serialization in an ascending or descending order.

::: caution

> While the above example is using a class it is recommended to stick with using `struct` when implementing `INetworkSerializable` due to its lower memory footprint and it helps to avoid other serialization related issues when using a class. 

Reviewer: ```suggestion
The example above uses a class, but it's generally recommended to use a `struct` when implementing `INetworkSerializable` due to its lower memory footprint. It also helps to avoid other serialization-related issues when using a class. 
```
